### ISSUE TITLE
cppstats as Python module

### ISSUE THREAD

**Issue by Ext3h** [clhunsen/cppstats#16]
Right now, cppstats is only callable via command line. However, since it's mostly written in Python, it would be favorable if it did not only provide a CLI but also an API for direct integration as a Python library into other projects.

This would require some abstraction between CLI parameters captured by ArgumentParser and the actual runtime configuration used in each pass.

This should primarily allow for better error handling when embedding cppstats into automated processes.

**Comment by clhunsen** [comment]
I totally agree, I will put it on the list. Although, the transformation of cppstats from an command-line and standalone tool towards a library might take some time.
